src/test/java/se/phooey/raining/web/IsItRainingE2ESeleniumTest.java

```diff
@@ -65,7 +66,7 @@ public void setUp() throws Exception {
 		profile.setPreference("geo.prompt.testing", true);
 		profile.setPreference("geo.prompt.testing.allow", true);
 		URL fileUrl = getClass().getClassLoader().getResource("spoofed_geolocation.json");
-		profile.setPreference("geo.wifi.uri", fileUrl.toExternalForm());
+		profile.setPreference("geo.wifi.uri", Objects.requireNonNull(fileUrl, "fileUrl is null").toExternalForm());
```

Comment:
Good change, this will make it more clear what went wrong if the file "spoofed_geolocation.json" cannot be found.

But the message could be more clear, e.g.:
final String fileName = "spoofed_geolocation.json";
profile.setPreference("geo.wifi.uri", Objects.requireNonNull(fileUrl, String.format("Cannot find file %s", fileName)).toExternalForm());

Reply:
Agreed!